src/main/java/com/amplitude/HttpTransport.java

```diff
@@ -304,40 +316,6 @@ private EventsRetryResult retryEventsOnce(String userId, String deviceId, List<E
     return eventsToRetry;
   }
 
-  private List<Event> pruneEvent(List<Event> events) {
```

Comment:
@yuhao900914 Can you confirm this is safe to remove?

Reply:
will take a look

Reply:
This function is for if we already have <user, device>  pair in the buffer, we directly put the event in the current retry.  
If we take the pruning method out, it will have some unnecessary compute in onEventsError later.

Reply:
This function cause two problems:
1. During the following process, events may be dropped based on invalid response body. These response contains index of events for original list. Events in pruned list got dropped by original list index cause index out of bound exception and we may drop wrong events.
2. During the following process, events may be dropped based on rate limit response body and event's userId/deviceId. The pruned list function skip this check and add events need to drop directly to retry buffer.